test/astar.cc

```diff
@@ -1791,6 +1791,267 @@ TEST(BiDiAstar, test_max_reserved_labels_count) {
   astar.Clear();
 }
 
+/*************************/
+
+class SimpleCost final : public valhalla::sif::DynamicCost {
+public:
+  // keep track of the time information with which edge cost was called
+  mutable std::vector<baldr::TimeInfo> time_infos;
```

Comment:
I've added a test for Bidir A* that makes sure passing different options causes different behavior. The setup is a bit manual and verbose, but that way we can record what `TimeInfo` objects are passed into `EdgeCost` by the algorithm: if the heuristic is used, we should never pass invalid time objects, if it's disabled we should expect _some_ of the passed time infos to be invalid.